website/content/docs/ecs/architecture.mdx

```diff
@@ -7,39 +7,53 @@ description: >-
+at startup and sets up initial configuration for Consul and Envoy.
 
 ### Task Startup
 
 This diagram shows the timeline of a task starting up and all its containers:
 
 ![Task Startup Timeline](/img/ecs-task-startup.png)
 
-- **T0:** ECS starts the task. The `discover-servers` container starts looking for the Consul server task’s IP.
-  It waits for the Consul server task to be running on ECS, looks up its IP and then writes the address to a file.
-  Then the container exits.
-- **T1:** Both the `consul-client` and `mesh-init` containers start:
-  - `consul-client` starts up and uses the server IP to join the cluster.
+- **T0:** ECS starts the task. The `consul-client` and `mesh-init` containers start:
+  - `consul-client` uses the `retry-join` option to join the Consul cluster
   - `mesh-init` registers the service for this task and its sidecar proxy into Consul. It runs `consul connect envoy -bootstrap` to generate Envoy’s bootstrap JSON file and write it to a shared volume. After registration and bootstrapping, `mesh-init` exits.
-- **T2:** The `sidecar-proxy` container starts. It runs Envoy by executing `envoy -c <path-to-bootstrap-json>`.
-- **T3:** The `sidecar-proxy` container is marked as healthy by ECS. It uses a health check that detects if its public listener port is open. At this time, the user’s application containers are started since all the Consul machinery is ready to service requests.
-- **T4:** Consul marks the service as healthy by running the health checks specified in the task Terraform. The service will now receive traffic. At this time the only running containers are `consul-client`, `sidecar-proxy` and the user’s application container(s).
+- **T1:** The `sidecar-proxy` container starts. It runs Envoy by executing `envoy -c <path-to-bootstrap-json>`.
+- **T2:** The `sidecar-proxy` container is marked as healthy by ECS. It uses a health check that detects if its public listener port is open. At this time, your application containers are started since all Consul machinery is ready to service requests. The only running containers are `consul-client`, `sidecar-proxy`, and your application container(s).
+
+### Automatic ACL Token Provisioning
+
+Consul ACL tokens secure communication between agents and services.
+The following containers in a task require an ACL token:
+
+- `consul-client`: The Consul client uses a token to authorize itself with Consul servers.
+  All `consul-client` containers share the same token.
+- `mesh-init`: The `mesh-init` container uses a token to register the service with Consul.
+  This token is unique for the Consul service, and is shared by instances of the service.
+
+The ACL controller automatically creates ACL tokens for mesh-enabled tasks in an ECS cluster.
+The `acl-controller` Terraform module creates the ACL token used by `consul-client` containers, and
+then starts the ACL controller task. The controller watches for tasks in the cluster. It checks tags
```

Comment:
Rephrasing a bit to clarify that the controller task will create the token rather than the module.
```suggestion
The `acl-controller` Terraform module the ACL controller task. The controller creates the ACL token used by `consul-client` containers at startup and then watches for tasks in the cluster. It checks tags
```